packages/tailwindcss/src/variants.ts

```diff
@@ -12,7 +12,7 @@ type VariantFn<T extends Variant['kind']> = (
 
 type CompareFn = (a: Variant, z: Variant) => number
 
-type CompoundKind = false | Array<'selector'>
+type CompoundKind = false | Array<'at-rule' | 'selector'>
```

Comment:
We discussed of using a bitmap here instead (this way we can save the countless array allocations)